src/app/services/auth.service.ts

```diff
@@ -0,0 +1,122 @@
+import { Injectable } from '@angular/core';
+import { Http, Headers, Response, RequestOptions } from '@angular/http';
+import { Router } from '@angular/router';
+import { Observable } from 'rxjs/Observable';
+import { environment } from '../../environments/environment';
+import 'rxjs/Rx';
+
+@Injectable()
+export class AuthService {
+
+	private static _localStorageKey = 'pcprepkitUser';
+	private _token: string = null; 
+	private _user: object = null;
+
+	// Routes
+	private _loginAuth = environment.baseURL + environment.authEndpoint +'login';
+	private _logoutAuth = environment.baseURL + environment.authEndpoint  + 'logout';
+	private _authenticatedApi = environment.baseURL + environment.authEndpoint  + 'authenticated';
+	private _forgotPasswordApi = environment.baseURL + environment.authEndpoint  + 'forgot';	
+	private _resetPasswordApi = environment.baseURL + environment.authEndpoint  + 'reset/';		
+
+	constructor(private _http: Http, private _router: Router) {
+	    const currentUser = JSON.parse(localStorage.getItem(AuthService._localStorageKey));
+	    if (currentUser && currentUser.user && currentUser.token) {
+			this._user = currentUser.user;
+			this._token = currentUser.token;
+	    } else {
+			localStorage.removeItem(AuthService._localStorageKey);
+	    }
+	}
+
+	public loginUser (body: Object): Observable<any> {
+		this.logout();
+	    let bodyString = JSON.stringify(body); 
+	    let headers      = new Headers({ 'Content-Type': 'application/json' }); 
+	    let options       = new RequestOptions({ headers: headers }); 
+
+	    return this._http.post(this._loginAuth, body, options) 
```

Comment:
i'm not very familiar with typescript, so i'll just javascript syntax to explain instead.

imagine an Api class/library:
```
class Api {
    constructor() {
    }

    getRequest(url, body, options) {
    }

    postRequest(url, body, options) {
    }

    putRequest(url, body, options) {
    }

    deleteRequest(url, body, options) {
    }
}
```

then, instead of calling `this._http.post(this._loginAuth, body, options)`, you can just call `Api.post(this._loginAuth, body, options)`.

since promises can be chained, you could implement the rejected/failed cases in the Api library, and handle each of the deferred/success cases separately in the individual calls